src/coord_seq.rs

```diff
@@ -149,6 +147,113 @@ impl<'a> CoordSeq<'a> {
         }
     }
 
+    /// Creates a new `CoordSeq` from an interleaved coordinate buffer.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::CoordSeq;
+    ///
+    /// let buffer = vec![0., 1., 2., 3., 4., 5.];
+    /// let coords = CoordSeq::new_from_buffer(&buffer, 3, 2, false, false)
+    ///                       .expect("failed to create CoordSeq");
+    /// assert_eq!(coords.get_y(1), Ok(3.));
+    /// assert_eq!(coords.get_x(2), Ok(4.));
+    /// ```
+    #[cfg(feature = "v3_10_0")]
+    pub fn new_from_buffer(
+        data: &[f64],
+        size: u32,
+        dims: u32,
+        has_z: bool,
+        has_m: bool,
+    ) -> GResult<CoordSeq<'a>> {
+        assert_eq!(
+            data.len(),
+            (size * dims) as usize,
+            "Incorrect buffer length"
+        );
+
+        let context_handle = ContextHandle::init_e(Some("CoordSeq::new_from_buffer"))?;
+        unsafe {
+            let ptr = GEOSCoordSeq_copyFromBuffer_r(
+                context_handle.as_raw(),
+                data.as_ptr(),
+                size as _,
+                has_z as _,
+                has_m as _,
+            );
+            CoordSeq::new_from_raw(
+                ptr,
+                Arc::new(context_handle),
+                size,
+                dims.into(),
+                "new_from_buffer",
+            )
+        }
+    }
+
+    /// Creates a new `CoordSeq` from separated coordinate buffers.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::CoordSeq;
+    ///
+    /// let x = vec![0., 2., 4.];
+    /// let y = vec![1., 3., 5.];
+    /// let coords = CoordSeq::new_from_arrays(&x, &y, None, None)
+    ///                       .expect("failed to create CoordSeq");
+    /// assert_eq!(coords.get_y(1), Ok(3.));
+    /// assert_eq!(coords.get_x(2), Ok(4.));
+    /// ```
+    #[cfg(feature = "v3_10_0")]
+    pub fn new_from_arrays(
+        x: &[f64],
+        y: &[f64],
+        z: Option<&[f64]>,
+        m: Option<&[f64]>,
+    ) -> GResult<CoordSeq<'a>> {
+        assert_eq!(x.len(), y.len(), "Arrays have different lengths.");
+
+        let mut dims: u32 = 2;
+        let z_ptr = if let Some(z) = z {
+            assert_eq!(x.len(), z.len(), "Arrays have different lengths.");
+            dims += 1;
+            z.as_ptr()
+        } else {
+            std::ptr::null()
+        };
+        let m_ptr = if let Some(m) = m {
+            assert_eq!(x.len(), m.len(), "Arrays have different lengths.");
+            dims += 1;
+            m.as_ptr()
+        } else {
+            std::ptr::null()
+        };
+
+        let size = x.len() as u32;
+
+        let context_handle = ContextHandle::init_e(Some("CoordSeq::new_from_arrays"))?;
+        unsafe {
+            let ptr = GEOSCoordSeq_copyFromArrays_r(
+                context_handle.as_raw(),
+                x.as_ptr(),
+                y.as_ptr(),
+                z_ptr,
+                m_ptr,
+                size as _,
```

Comment:
So as mentioned above, you can directly do:

```suggestion
                x.len() as _,
```